src/biogeochem/FanMod.F90

```diff
@@ -0,0 +1,1214 @@
+    ! should be small.
+    garbage = flux_out(size(mtan)) * dt
+    
+  end subroutine age_pools_soil
+
+  subroutine age_pools_slurry(ndep, dt, water_slurry, tan_slurry, tan_soil, pools, garbage)
+    !
+    ! Age pools, modified for the 4-stage slurry model (only below-surface slurry ages).
+    !
+    implicit none
+    real(r8), intent(in) :: ndep ! flux of TAN input, gN/m2/s
+    real(r8), intent(in) :: dt ! timestep, s
+    ! water in slurry pool, on surface (1) and below surface (2) not including the background water content (theta)
+    real(r8), intent(in) :: water_slurry(:)
+    real(r8), intent(inout) :: tan_slurry  ! TAN in slurry pool, gN/m2
+    real(r8), intent(inout) :: tan_soil(:) ! TAN in soil pools, gN/m2
+    real(r8), intent(in) :: pools(:) ! age spans covered by each pool, including the S0 surface slurry. seconds.
+    real(r8), intent(out) :: garbage       ! garbage TAN, see above
+
+    real(r8) :: fract_slurry_soil, flux_to_soilpools
+    
+    fract_slurry_soil = water_slurry(2) / sum(water_slurry)
+    flux_to_soilpools = tan_slurry * fract_slurry_soil / pools(1)
+    tan_slurry = tan_slurry + (ndep - flux_to_soilpools) * dt
+
+    call age_pools_soil(flux_to_soilpools, dt, pools(2:), tan_soil, garbage)
+    
+  end subroutine age_pools_slurry
+
+  !************************************************************************************
+  ! Public functions for integrating the FAN model for one timestep.
+  
+  subroutine update_4pool(tg, ratm, theta, thetasat, precip, evap, qbot, watertend, runoff, tandep, tanprod, bsw, &
+       depth_slurry, poolranges, tanpools, Hconc, fluxes, residual, dt, dz_layer, pools_size, fluxes_size, status)
+    !
+    ! Evaluate fluxes and integrate states for a 4-stage slurry model with first pool
+    ! representing uninfiltrated slurry.
+    !
+    implicit none
+    real(r8), intent(in) :: tg ! soil temperature, K
+    real(r8), intent(in) :: ratm ! atmospheric resistance, s/m
+    real(r8), intent(in) :: theta ! volumetric soil water in soil column (unaffected by slurry)
+    real(r8), intent(in) :: thetasat ! vol. soil water at saturation
+    real(r8), intent(in) :: precip   ! precipitation, m/s
+    real(r8), intent(in) :: evap   ! ground evaporation, m/s
+    real(r8), intent(in) :: qbot   ! specific humidity (kg/kg) at lowest atmospheric model level
+    real(r8), intent(in) :: watertend ! time derivative of theta
+    real(r8), intent(in) :: runoff    ! surface runoff flux, m/s
+    real(r8), intent(in) :: tandep    ! TAN input flux, gN/m2/s
+    real(r8), intent(in) :: tanprod   ! TAN produced in the column, added to aged TAN pool
+    real(r8), intent(in) :: bsw
+    real(r8), intent(in) :: depth_slurry ! Initial slurry depth, m
+    ! age ranges of TAN pools S0, S1, S2, S3 sec. Slurry infiltration time is inferred from S0. 
+    integer, intent(in) :: pools_size ! size of the tanpools array. >= 4
+    real(r8), intent(in) :: poolranges(pools_size)
```

Comment:
Here and elsewhere: It's safer to declare arrays to be assumed size (`array_name(:)`) then add a SHR_ASSERT call that asserts that the array is the expected size. Otherwise, Fortran compilers will let you pass incorrectly-sized arrays in.

Reply:
Thanks for fixing the declarations. (It doesn't look like you added assertions on the array sizes, but that doesn't seem critical here, so I'm marking this as resolved.)